openedx_rocketchat/utils.py

```diff
@@ -0,0 +1,205 @@
+
+
+LOG = logging.getLogger(__name__)
+CACHE_TIMEOUT = get_configuration_helpers().get_value("ROCKET_CHAT_CACHE_TIMEOUT", 86400)
+ROCKET_CHAT_DATA = "rocket_chat_data"
+
+
+def get_rocket_chat_settings():
+    """
+    Returns a dict with rocket chat settings. This try to get form the general settings and xblock settings.
+    **Example**:
+
+        ROCKET_CHAT_SETTINGS = {
+            'private_url_service': 'http://localhost:3000',
+            'public_url_service': 'localhost:3000',
+            'admin_user': 'rocketchat_admin_user',
+            "admin_pass": 'pass1234',
+            "username":"staff",
+            "password":"edx",
+        }
+
+        XBLOCK_SETTINGS = {
+            'RocketChatXBlock': {
+                'private_url_service': 'http://localhost:3000',
+                'public_url_service': 'localhost:3000',
+                'admin_user': 'rocketchat_admin_user',
+                "admin_pass": 'pass1234',
+                "username":"staff",
+                "password":"edx",
+            }
+        }
+    """
+    try:
+        return settings.ROCKET_CHAT_SETTINGS
+    except AttributeError, settings_error:
+        LOG.warning('Get settings warning: %s', settings_error)
+        pass
+    try:
+        xblock_settings = settings.XBLOCK_SETTINGS
+        return xblock_settings.get('RocketChatXBlock', None)
+    except AttributeError, xblock_settings_error:
+        LOG.warning('Get settings warning: %s', xblock_settings_error)
+        pass
+
+    return {}
+
+
+def create_user(api_rocket_chat, user, course_key):
+    """
+    Create a user in rocketChat
+    """
+    user, u_prof = get_user(user.email)
+    anonymous_id = anonymous_id_for_user(user, course_key)
+    return api_rocket_chat.users_create(
+        email=user.email,
+        name=u_prof.name if u_prof.name != "" else user.username,
+        password=anonymous_id,
+        username=user.username
+    )
+
+
+def create_course_group(api_rocket_chat, course_id, user_id, username):
+    """
+    Add a user to the course group
+    """
+    course = re.sub('[^A-Za-z0-9]+', '', course_id)
+    room_name = "{}__{}".format("General", course)
+    response = api_rocket_chat.groups_info(room_name=room_name)
+    try:
+        response = response.json()
+
+        if response['success']:
+            api_rocket_chat.groups_invite(response['group']['_id'], user_id)
+        else:
+            kwargs = {
+                'members': [username],
+                'customFields': {
+                    'course': course,
+                    'team': None,
+                    'topic': None,
+                    'specificTeam': False,
+                }
+            }
+            api_rocket_chat.groups_create(name=room_name, **kwargs)
+
+    except AttributeError:
+        LOG.error("Create Course Group error: response with status code = %s", response.status_code)
+        pass
```

Comment:
Why do you are catching this error if you are going to 'pass' anyway? @andrey-canon

Reply:
ok